ref/general/authentication.adoc

```diff
@@ -13,16 +13,20 @@
 
-image::/docs/img/Authentication.png[Authentication diagram,align="center"]
+image::/docs/img/Authentication-rvsd.png[Authentication diagram,align="center"]
 
 == User registries
-A user registry is a database that contains information about a set of users and their associated roles. When a user requests a protected resource in an application, the user's credentials are checked against the information in one or more user registries. Open Liberty provides an easy-to-use link:/docs/ref/general/#basic-registry.html[basic user registry] for developers to use in test environments. Typically, application servers are configured to use an external registry, such as a link:/docs/ref/general/#LDAP-registry.html[Lightweight Directory Access Protocol (LDAP) registry]. Open Liberty also supports federated registries so that a server can be configured to authenticate from multiple registries.
+A user registry is a database that contains information about a set of users and their associated roles. When a user requests a protected resource in an application, the user's credentials are checked against the information in one or more user registries. Open Liberty provides an easy-to-use link:/docs/ref/general/#basic-registry.html[basic user registry] for developers to use in test environments. Typically, applications in production are configured to use an external registry, such as a link:/docs/ref/general/#LDAP-registry.html[Lightweight Directory Access Protocol (LDAP) registry].
+
+Open Liberty also supports link:/docs/ref/general/#federated-registries.html[federated user registries] so that a server can be configured to authenticate from multiple registries from a single logical view. Information from LDAP, basic, and custom user registries can be federated into a common registry to manage application security.
+
+If your application needs to reference a database other than a basic or LDAP registry, you can configure a custom registry by implementing the `UserRegistry` interface. The `UserRegistry` interface is a Service Programming Interface (SPI) that enables support to virtually any type of account repository.
```

Comment:
User Registry is an API. The S in SPI typically means System, not Service. What about using data store over database?

Reply:
The UserRegistry/SPI info here comes from this KC page : https://www.ibm.com/support/knowledgecenter/SSAW57_liberty/com.ibm.websphere.wlp.nd.multiplatform.doc/ae/twlp_sec_custmr.html- 

which is basically implement SPI>convert to OSIG & bundle>install as usr feature.

If there's a User Registry API we should doc instead, or in addition to, is there autogen doc for it"? I can't find it but maybe I'm looking in the wrong places

For now, just replacing SPI w/ API

Reply:
I think replacing SPI with API in this instance is fine. UserRegistry has duel roles as both an SPI and an API, so in this case I think just going simple with API is fine.

The doc on writing OSGi bundle is out of date and no longer required for UserRegistry. It doesn't massively change the content in this document, but if we get to writing custom UserRegistry we should document the simpler approach.